Source/JavaScriptCore/jit/JITPropertyAccess.cpp

```diff
@@ -1660,16 +1691,51 @@ MacroAssemblerCodeRef<JITThunkPtrTag> JIT::slow_op_resolve_scopeGenerator(VM& vm
     Jump exceptionCheck = jit.jump();
 
     LinkBuffer patchBuffer(jit, GLOBAL_THUNK_ID, LinkBuffer::Profile::Thunk);
-    patchBuffer.link(operation, FunctionPtr<OperationPtrTag>(operationResolveScopeForBaseline));
+    patchBuffer.link(operation, FunctionPtr<OperationPtrTag>(operationFunction));
     patchBuffer.link(exceptionCheck, CodeLocationLabel(vm.getCTIStub(checkExceptionGenerator).retaggedCode<NoPtrTag>()));
     return FINALIZE_THUNK(patchBuffer, JITThunkPtrTag, "Baseline: slow_op_resolve_scope");
 }
 
-void JIT::emit_op_get_from_scope(const JSInstruction* currentInstruction)
+MacroAssemblerCodeRef<JITThunkPtrTag> JIT::slow_op_resolve_scopeGenerator(VM& vm)
 {
-    auto bytecode = currentInstruction->as<OpGetFromScope>();
+    return slow_op_resolve_scopeGenerator_helper<OpResolveScope>(vm, operationResolveScopeForBaseline);
+}
+
+#define GET_FROM_SCOPE_CODE_REF(thunk)                                             \
```

Comment:
ditto, can this be a function instead of a macro?

Reply:
ditto, are two functions better?